extra/to_movement_ops.py

```diff
@@ -0,0 +1,107 @@
+import random
+from tqdm import tqdm
+from extra.optimization.helpers import load_worlds
+from tinygrad.codegen.linearizer import Linearizer
+from tinygrad.ops import LazyOp, MovementOps, TernaryOps, BinaryOps, UnaryOps, ReduceOps, BufferOps, MemBuffer, ConstBuffer
+from tinygrad.helpers import dtypes, prod
+from tinygrad.shape.shapetracker import ShapeTracker
+from tinygrad.shape.view import View
+from tinygrad.shape.symbolic import Node, Variable
+inf, nan = float('inf'), float('nan')
+
+def get_buffer_size(shape, strides, offset, mask):
+  real_shape = tuple(y-x for x,y in mask) if mask else shape
+  offset = offset + sum(strides[i] * (real_shape[i]-1) for i in range(len(strides)) if strides[i]<0)
+  real_offset = offset + (sum(x*st for (x,_),st in zip(mask, strides)) if mask else 0)
+  real_real_shape = [s for s,st in zip(real_shape, strides) if st]
+  strides = [abs(st) for st in strides if st and not isinstance(st, Node)]
+  return real_offset + sum((s-1)*st for s, st in zip(real_real_shape,strides)) + 1
+
+def flatten_view(view: View):
+  real_shape = tuple(y-x for x,y in view.mask) if view.mask else view.shape
+  offset = view.offset + sum(view.strides[i] * (real_shape[i]-1) for i in range(len(view.strides)) if view.strides[i]<0)
```

Comment:
`zip`. Come to think of it, the first lines of this and the preceding function are the same.